torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -676,13 +676,47 @@ def rewrite_node(node):
             if node.target == torch.ops.aten._to_copy.default:
                 target_users_ops = [user.target for user in node.users.keys()]
                 # Float and int types can be converted to ttnn.bfloat16, but bool may be problematic
+                # Can be removed if bool is only used for the following ops that have lowering:
+                ops_safe_to_remove = set([torch.ops.aten.masked_fill.Scalar])
+                can_remove_bool = kwargs["dtype"] != bool or (
+                    kwargs["dtype"] == torch.bool and ops_safe_to_remove.intersection(target_users_ops)
+                )
                 # Skip if type casting from bool and if the graph output uses this op
-                if kwargs["dtype"] not in [torch.bool] and "output" not in target_users_ops:
+                if can_remove_bool and "output" not in target_users_ops:
```

Comment:
Does not look like a part of masked_fill change

Reply:
This is dependent on masked_fill, but can be separated into its own PR. I'll revert this part.